backend/src/Prelude/Prelude.fs

```diff
@@ -256,13 +257,23 @@ module NEList =
     | [] -> Exception.raiseInternal msg metadata
     | head :: tail -> { head = head; tail = tail }
 
-
-
   let ofSeq (head : 'a) (seq : seq<'a>) : NEList<'a> =
     { head = head; tail = List.ofSeq seq }
 
   let singleton (head : 'a) : NEList<'a> = { head = head; tail = [] }
 
+  let push (l : NEList<'a>) (head : 'a) : NEList<'a> =
+    { head = head; tail = l.head :: l.tail }
+
+  let reverse (l : NEList<'a>) : NEList<'a> =
+    match toList l with
```

Comment:
You can access `l.head` and `l.tail` directly, which I think would make this code nicer than using `toList`